modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -83,6 +83,86 @@ export interface SiteConfig<Config> {
     config: Config;
 }
 
+export type DefaultHeaders = {
+    Accept: string;
+    'Accept-Charset': string;
+    'Accept-Encoding': string;
+    Authorization: string;
+    'Content-Length': string,
+    'Content-Type': string,
+    Cookies: string;
+    Expect: string;
+    Host: string;
+    'If-None-Match': string;
+    Language: string;
+    'User-Agent': string;
+};
+
+export interface Cookie {
+    value: string;
+    path?: string;
+    domain?: string;
+    comment?: string;
+    maxAge?: number; // -1
+    secure?: boolean; // false
+    httpOnly?: boolean; // false
+    sameSite?: 'lax' | 'strict' | 'none' | ''; // ''
+}
+
+interface Request<Params extends Record<string, string | undefined> = Record<string, string | undefined>,
```

Comment:
If you have a url with the same query parameter listed more then once, the value of the param in XP is an `Array<string>`

```suggestion
interface Request<Params extends Record<string, string | Array<string> | undefined> = Record<string, string | Array<string> | undefined>,
```

Reply:
I think it's better to not pass in these type parameters, because your `get` or `post` function has to be able to handle any input, both missing and string and Arrays, so there is little value in locking in a shape.

The value of specifying these shapes in type parameters, would be if you could use the same shape "in both ends". That means that you specify an `interface` that can be used both by `Request` and `serviceUrl()` or `pageUrl()`.